File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int maxi=-1e9, mini=1e9;
        for(int it:prices){
            mini = min(mini, it);
            maxi = max(maxi,it-mini);
        }
        return maxi;

    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int ans = 0;
        int minPrice=Integer.MAX_VALUE;
        for(int i=0;i<n;i++){
            minPrice = Math.min(minPrice, prices[i]);
            ans= Math.max(ans,prices[i]-minPrice);
        }
        return ans;
    }
}